mode/gap/gap.js

```diff
@@ -0,0 +1,207 @@
+      }
+       // Handle operators and Delimiters
+        if (stream.match(doubleDelimiters))
+            return "punctuation";
+
+        if (stream.match(doubleOperators) || stream.match(singleOperators))
+            return "operator";
+
+        if (stream.match(singleDelimiters))
+            return "punctuation";
+
+        if (state.lastToken == "." && stream.match(identifiers))
+            return "property";
+
+        if (stream.match(keywords) || stream.match(wordOperators))
+            return "keyword";
+
+        if (stream.match(builtins))
+            return "builtin";
+
+        if (stream.match(identifiers)) {
+            return "variable";
+        }
+
+      // Handle Number Literals
+      if (stream.match(/^[0-9\.]/, false)) {
+        var floatLiteral = false;
+        // Floats
+        if (stream.match(/^\d*\.\d+(e[\+\-]?\d+)?/i)) { floatLiteral = true; }
+        if (stream.match(/^\d+\.\d*/)) { floatLiteral = true; }
+        if (stream.match(/^\.\d+/)) { floatLiteral = true; }
+        if (floatLiteral) {
+          return "number";
+        }
+        // Integers
+        if (stream.match(/^\d+/)) {
+          return "number";
+        }
+      }
+
+      // Literal permutation. Note that permutations do not
+      // necessarily come as literals.
+      if (stream.match(/^\([1-9][0-9]*(,[1-9][0-9]*)*\)/)) {
+          return "number";
+      }
+
+      // Handle Strings
+      if (stream.match(stringPrefixes)) {
+        state.tokenize = tokenStringFactory(stream.current());
+        return state.tokenize(stream, state);
+      }
+
+     // Handle non-detected items
+      stream.next();
+      return ERRORCLASS;
+    }
+
+    function tokenStringFactory(delimiter) {
+      var singleline = delimiter.length == 1;
+      var OUTCLASS = "string";
+
+    function tokenString(stream, state) {
+      while (!stream.eol()) {
+          stream.eatWhile(/[^'"\\]/);
+          if (stream.eat("\\")) {
+            stream.next();
+            if (singleline && stream.eol())
+              return OUTCLASS;
+          } else if (stream.match(delimiter)) {
+            state.tokenize = tokenBase;
+            return OUTCLASS;
+          } else {
+            stream.eat(/['"]/);
+          }
+        }
+        if (singleline) {
+          if (parserConf.singleLineStringErrors)
+            return ERRORCLASS;
+          else
+            state.tokenize = tokenBase;
+        }
+        return OUTCLASS;
+      }
+      tokenString.isString = true;
+      return tokenString;
+    }
+
+    function tokenLexer(stream, state) {
+      var style = state.tokenize(stream, state);
+
+      if ((style == "variable" || style == "builtin")
+          && state.lastToken == "meta")
+        style = "meta";
+
+      return style;
+    }
+
+    var external = {
+      startState: function(basecolumn) {
+        return {
+          tokenize: tokenBase,
+          scopes: [{offset: basecolumn || 0, type: "g", align: null}],
```

Comment:
Since you're never updating the `scopes` property, you might just remove it (along with the `indent` method, which doesn't work in the current code.)